app/models/user.rb

```diff
@@ -183,7 +183,11 @@ def remove_from_access_group(gid)
     save!(:validate => false)
   end
 
-    private
+  def after_aaf_rc_authentication
+    raise Exception.new('Unauthorized') unless approved?
```

Comment:
Its better to raise a subclass of StandardError rather than Exception. I just to do this too :)

Exception is too general and could include errors like me NoMemoryError which generally you don't catch but just let the application crash.